#! /bin/sh /usr/share/dpatch/dpatch-run
## 09_trailingspace.dpatch by  <az@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: when copying tags from v1, remove trailing spaces.

@DPATCH@
diff -urNad mp3gain-1.5.1~/id3tag.c mp3gain-1.5.1/id3tag.c
--- mp3gain-1.5.1~/id3tag.c	2009-06-29 14:01:34.000000000 +0200
+++ mp3gain-1.5.1/id3tag.c	2010-01-10 08:21:41.272091362 +0100
@@ -823,7 +823,7 @@
 static int id3_parse_v1_tag(FILE *f, struct ID3v2TagStruct *tag)
 {
 	unsigned char buf[128];
-	char sbuf[32];
+	char sbuf[32],*p;
 
 	struct ID3v2FrameStruct **pframe;
 
@@ -850,6 +850,8 @@
 	if (buf[3] != '\0') {
 		memcpy(sbuf, buf + 3, 30);
 		sbuf[30] = '\0';
+		/* get rid of trailing spaces */
+		for(p=sbuf+29; *p==' ' && p>=sbuf; *p--='\0');
 		*pframe = id3_make_frame("TIT2", "bs", 0, sbuf);
 		pframe = &((*pframe)->next);
 	}
@@ -858,6 +860,9 @@
 	if (buf[33] != '\0') {
 		memcpy(sbuf, buf + 33, 30);
 		sbuf[30] = '\0';
+		/* get rid of trailing spaces */
+		for(p=sbuf+29; *p==' ' && p>=sbuf; *p--='\0');
+		DBG(("fixed v1 artist: \"%s\"\n",sbuf));
 		*pframe = id3_make_frame("TPE1", "bs", 0, sbuf);
 		pframe = &((*pframe)->next);
 	}
@@ -866,6 +871,8 @@
 	if (buf[63] != '\0') {
 		memcpy(sbuf, buf + 63, 30);
 		sbuf[30] = '\0';
+		/* get rid of trailing spaces */
+		for(p=sbuf+29; *p==' ' && p>=sbuf; *p--='\0');
 		*pframe = id3_make_frame("TALB", "bs", 0, sbuf);
 		pframe = &((*pframe)->next);
 	}
@@ -885,6 +892,8 @@
 	if (buf[97] != '\0') {
 		memcpy(sbuf, buf + 97, 30);
 		sbuf[30] = '\0';
+		/* get rid of trailing spaces */
+		for(p=sbuf+29; *p==' ' && p>=sbuf; *p--='\0');
 		/* assume ISO-8859-1, unknown language, no description */
 		*pframe = id3_make_frame("COMM", "bssbs", 0, "XXX", "", 0, sbuf);
		pframe = &((*pframe)->next);
